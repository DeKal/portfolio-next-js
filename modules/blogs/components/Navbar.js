import React, { Fragment } from 'react'
import Container from 'react-bootstrap/Container'
import Navbar from 'react-bootstrap/Navbar'
import Nav from 'react-bootstrap/Nav'
import { Translate } from 'react-localize-redux'
import NavLogo from '~/modules/blogs/components/navbar/NavLogo'
import NavItem from '~/modules/blogs/components/navbar/NavItem'
import Search from '~/modules/blogs/components/navbar/Search'
import { PAGE_LIST } from '~/modules/blogs/consts/pages'
import { REDIRECT_LINK, LOGO_IMAGE } from '~/modules/blogs/consts/logo'

const NavBar = () => (
  <Fragment>
    <Navbar bg="light" variant="light">
      <Container>
        <NavLogo redirectLink={REDIRECT_LINK} image={LOGO_IMAGE} />
        <Nav className="mr-auto">
          {PAGE_LIST.map(page => (
            <NavItem key={page.name} href={page.url}>
              <Translate id={page.name} />
            </NavItem>
          ))}
        </Nav>
        <Search />
      </Container>
    </Navbar>
  </Fragment>
)

export default NavBar
