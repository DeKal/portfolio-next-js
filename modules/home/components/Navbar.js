import React, { useState } from 'react'

import styled from 'styled-components/macro'
import NavHeader from '~/modules/home/components/nav/NavHeader'
import NavContent from '~/modules/home/components/nav/NavContent'
import { INIT_NAV_CONTENT_SHOW } from '~/modules/home/consts/initState'

const Navbar = () => {
  const [isShowNavContent, switchNav] = useState(INIT_NAV_CONTENT_SHOW)

  return (
    <MainHeader>
      <NavHeader isShowNavContent={isShowNavContent} switchNav={switchNav} />
      <NavContent isShowNavContent={isShowNavContent} />
    </MainHeader>
  )
}

const MainHeader = styled.header`
  position: fixed;
  z-index: 100;
  width: 100vw;
  height: ${props => props.theme.navHeightWhenScrollingOnMobile};
`

export default Navbar
