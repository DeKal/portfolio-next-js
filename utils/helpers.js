import { INIT_SELECTED_PAGE } from '~/consts/initState'
import $ from 'jquery'

export const initHashLocation = () => {
  if (location.hash) {
    const offset = $(location.hash).offset()
    window.scrollTo(offset.left, offset.top)
  }
}

export const isSafari = () => {
  var userAgent = navigator.userAgent.toLowerCase()
  if (userAgent.indexOf('safari') != -1) {
    // Safari agent will not have Chrome
    if (userAgent.indexOf('chrome') > -1) {
      return false
    } else {
      return true
    }
  }
}

export const setJqueryScrollEvent = () => {
  $('a').on('click', function(event) {
    // Make sure this.hash has a value before overriding default behavior
    if (this.hash) {
      // Prevent default anchor click behavior
      event.preventDefault()
      // Store hash
      var hash = this.hash

      $('html, body').animate(
        {
          scrollTop: $(hash).offset().top
        },
        400,
        function() {
          // Add hash (#) to URL when done scrolling (default click behavior)
          window.location.hash = hash
        }
      )
    } // End if
  })
}

export const cleanUpScrollEvent = () => {
  $('a').unbind()
}

export const getSelectedSection = () => {
  if (location.hash.length !== 0) {
    const selectedPage = location.hash.substr(1)
    return selectedPage
  }
  return INIT_SELECTED_PAGE
}

export const isChristmasDay = () => {
  const today = new Date()
  const month = today.getMonth() + 1
  const isDecember = month == 12
  const day = today.getDate()
  const isChristMasTime = 20 <= day <= 30

  return isDecember && isChristMasTime
}

export const isNewYearComing = () => {
  const today = new Date()
  const day = today.getDate()
  const isFirstTwoWeek = 1 <= day <= 16
  const month = today.getMonth() + 1
  const isJanuary = month == 1

  return isJanuary && isFirstTwoWeek
}
