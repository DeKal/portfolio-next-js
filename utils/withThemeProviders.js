import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import defaultTheme from '~/views/theme'

export const shallowWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

export const mountWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}
