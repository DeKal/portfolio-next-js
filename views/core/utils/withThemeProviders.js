import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import homeTheme from '~/theme/theme'
import baseTheme from '~/theme/baseTheme'

/* istanbul ignore next */
export const shallowWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

/* istanbul ignore next */
export const shallowWithDivingTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children).dive()
}

/* istanbul ignore next */
export const mountWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}

/* istanbul ignore next */
export const shallowWithBaseTheme = (children, theme = baseTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}
