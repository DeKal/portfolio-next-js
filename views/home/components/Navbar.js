import React, { useState } from 'react'

import styled from 'styled-components/macro'
import NavHeader from '~/views/home/components/nav/NavHeader'
import NavContent from '~/views/home/components/nav/NavContent'
import { INIT_NAV_CONTENT_SHOW } from '~/views/home/consts/initState'

const Navbar = () => {
  const [isShowNavContent, switchNav] = useState(INIT_NAV_CONTENT_SHOW)

  return (
    <MainHeader>
      <NavHeader isShowNavContent={isShowNavContent} switchNav={switchNav} />
      <NavContent isShowNavContent={isShowNavContent} />
    </MainHeader>
  )
}

const MainHeader = styled.header`
  position: fixed;
  z-index: 100;
  width: 100vw;
  height: ${props => props.theme.navHeightWhenScrollingOnMobile};
`

export default Navbar
