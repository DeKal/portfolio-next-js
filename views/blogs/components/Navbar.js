import React, { Fragment } from 'react'
import Container from 'react-bootstrap/Container'
import Navbar from 'react-bootstrap/Navbar'
import Nav from 'react-bootstrap/Nav'
import NavLogo from '~/views/blogs/components/navbar/NavLogo'
import NavItem from '~/views/blogs/components/navbar/NavItem'
import Search from '~/views/blogs/components/navbar/Search'

const NavBar = () => (
  <Fragment>
    <Navbar bg="light" variant="light">
      <Container>
        <NavLogo redirectLink="/blogs" image="/images/logo.png" />
        <Nav className="mr-auto">
          <NavItem href="/">Home</NavItem>
          <NavItem href="/blogs">Blogs</NavItem>
        </Nav>
        <Search />
      </Container>
    </Navbar>
  </Fragment>
)

export default NavBar
