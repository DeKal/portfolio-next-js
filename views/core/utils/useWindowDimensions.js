import { useState, useEffect } from 'react'
import { getWindowDimensions } from '~/views/core/utils/helpers'

/* istanbul ignore next */
const useWindowDimensions = () => {
  const [windowDimensions, setWindowDimensions] = useState(
    getWindowDimensions()
  )

  useEffect(() => {
    /* istanbul ignore next */
    function handleResize() {
      setWindowDimensions(getWindowDimensions())
    }

    window.addEventListener('resize', handleResize)
    /* istanbul ignore next */
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  return windowDimensions
}

export default useWindowDimensions
