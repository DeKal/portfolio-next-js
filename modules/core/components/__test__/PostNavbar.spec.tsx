import * as React from 'react'
import { shallowWithTheme } from '~/modules/core/utils/withThemeProviders'
import Navbar from '~/modules/core/components/PostNavbar'

describe('Post Navbar', () => {
  it('should render without throwing an error', function () {
    const wrap = shallowWithTheme(<Navbar />)
    expect(wrap).toMatchSnapshot()
  })
})
