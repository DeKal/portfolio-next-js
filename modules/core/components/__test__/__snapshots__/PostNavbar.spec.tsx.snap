// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Navbar should render without throwing an error 1`] = `
<Fragment>
  <Navbar
    bg="light"
    collapseOnSelect={false}
    expand={true}
    variant="light"
  >
    <Container
      fluid={false}
    >
      <NavLogo
        image="/images/logo.png"
        redirectLink="/blogs"
      />
      <Nav
        className="mr-auto"
        fill={false}
        justify={false}
      >
        <NavItem
          href="/"
          key="Home"
        >
          <Translate
            id="Home"
          />
        </NavItem>
        <NavItem
          href="/blogs"
          key="Blogs"
        >
          <Translate
            id="Blogs"
          />
        </NavItem>
      </Nav>
      <Search />
    </Container>
  </Navbar>
</Fragment>
`;
